mist/runtime/soft_skeleton.py

```diff
@@ -0,0 +1,50 @@
+
+import torch
+import torch.nn.functional as F
+
+class SoftSkeletonize(torch.nn.Module):
+
+    def __init__(self, num_iter=40):
+
+        super(SoftSkeletonize, self).__init__()
+        self.num_iter = num_iter
+
+    def soft_erode(self, img):
+
+        if len(img.shape)==4:
```

Comment:
We won't ever encounter this case since MIST only deals with 3D images. I'd remove this. Also add an else statement that throws an error if len(img.shape) does not equal 5. Something like:

else:
  raise ValueError("[Loss Error] len(img.shape) is not equal to 5")